static/js/reactions.js

```diff
@@ -127,13 +127,15 @@ exports.add_reaction = function (event) {
         }
         event.count = 1;
         event.title = new_title;
+        event.class = "message_reaction reacted";
         var reaction_button_element = message_reactions_element.find('.reaction_button');
         $(templates.render('message_reaction', event)).insertBefore(reaction_button_element);
     } else {
         var reaction = message_reactions_element.find("[data-emoji-name='" + event.emoji_name + "']");
         var count_element = reaction.find('.message_reaction_count');
         count_element.html(user_list.length);
         reaction.prop('title', new_title);
+        reaction.addClass("reacted");
```

Comment:
This function is called regardless from `server_events.js` to update the display, regardless of which user reacted, so you need a conditional on user_id here (and same for remove_reaction, I think).

I caught this by testing using 2 browsers open logged into different users; I recommend you use that testing process for this feature.